src/vellum/workflows/nodes/core/map_node/node.py

```diff
@@ -104,19 +107,36 @@ def run(self) -> Iterator[BaseOutput]:
                 subworkflow_event = map_node_event[1]
                 self._context._emit_subworkflow_event(subworkflow_event)
 
-                if subworkflow_event.name == "workflow.execution.initiated":
+                if (
+                    subworkflow_event.name == "workflow.execution.initiated"
+                    and subworkflow_event.workflow_definition == self.subworkflow
+                ):
                     for output_name in mapped_items.keys():
                         yield BaseOutput(name=output_name, delta=(None, index, "INITIATED"))
 
-                elif subworkflow_event.name == "workflow.execution.fulfilled":
-                    workflow_output_vars = vars(subworkflow_event.outputs)
-
-                    for output_name in workflow_output_vars:
-                        output_mapped_items = mapped_items[output_name]
-                        output_mapped_items[index] = workflow_output_vars[output_name]
+                elif (
+                    subworkflow_event.name == "workflow.execution.fulfilled"
+                    and subworkflow_event.workflow_definition == self.subworkflow
+                ):
+                    for output_reference, output_value in subworkflow_event.outputs:
```

Comment:
the other changes here were unrelated, but should build some more resilience and avoid use of `vars` since our outputs class is iterable